soccer/Robot.cpp

```diff
@@ -587,12 +618,51 @@ void OurRobot::replanIfNeeded(const Geometry2d::CompositeShape& global_obstacles
 	if (!_pathInvalidated) {
 		addText("Reusing path", Qt::white, "Planning");
 	} else {
-		Planning::Path *path = _planner->run(pos, angle, vel, _motionConstraints, &full_obstacles);
-		
+		double leadTime = *(_motionConstraints._replan_lead_time) * 1000000;
+		RobotPose predictedPose;
+
+		filter()->predict(timestamp() + leadTime, &predictedPose);
+		std::unique_ptr<Planning::Path> path = nullptr;
+		int count = 0;
+		while (!path) {
```

Comment:
Its to retry pathplanning if the pathplanner failed. It'll be unnecessary once we fix pathplanning and I think I stopped it form erroring out quite as much now.